runconfig/hostconfig.go

```diff
@@ -2,11 +2,39 @@ package runconfig
 
 import (
 	"encoding/json"
+	"fmt"
 	"io"
 
 	"github.com/docker/docker/api/types/container"
 )
 
+var (
+	// ErrConflictContainerNetworkAndLinks conflict between --net=container and links
+	ErrConflictContainerNetworkAndLinks = fmt.Errorf("Conflicting options: container type network can't be used with links. This would result in undefined behavior")
```

Comment:
They are mostly used in `runconfig/hostconfig_unix.go`, which is why I had them in hostconfig, but `errors.go` makes sense. I'll move them